stdlib/tests/crypto/stark/verifier_recursive/mod.rs

```diff
@@ -91,18 +93,23 @@ pub fn generate_advice_inputs(
 
     // read the main and auxiliary segments' OOD frames and add them to advice tape
     let ood_trace_frame = channel.read_ood_trace_frame();
-    let _ood_main_trace_frame = ood_trace_frame.main_frame();
-    let _ood_aux_trace_frame = ood_trace_frame.aux_frame();
-
-    let mut main_and_aux_frame_states = Vec::new();
-    for col in 0.._ood_main_trace_frame.current().len() {
-        main_and_aux_frame_states.push(_ood_main_trace_frame.current()[col]);
-        main_and_aux_frame_states.push(_ood_main_trace_frame.next()[col]);
-    }
-    for col in 0.._ood_aux_trace_frame.as_ref().unwrap().current().len() {
-        main_and_aux_frame_states.push(_ood_aux_trace_frame.as_ref().unwrap().current()[col]);
-        main_and_aux_frame_states.push(_ood_aux_trace_frame.as_ref().unwrap().next()[col]);
-    }
+    let ood_main_trace_frame = ood_trace_frame.main_frame();
+    let ood_aux_trace_frame = ood_trace_frame.aux_frame();
+
+    let mut main_and_aux_frame_states = ood_main_trace_frame.current().to_vec();
+    main_and_aux_frame_states.extend_from_slice(
+        ood_aux_trace_frame
+            .as_ref()
+            .expect("execution trace should have an auxiliary segment")
+            .current(),
+    );
+    main_and_aux_frame_states.extend_from_slice(ood_main_trace_frame.next());
+    main_and_aux_frame_states.extend_from_slice(
+        ood_aux_trace_frame
+            .as_ref()
+            .expect("execution trace should have an auxiliary segment")
+            .next(),
+    );
```

Comment:
Could we add a small comment here explaining what arrangement we are trying to achieve? I think it is something like:
```
[main_current_elements, aux_current_elements, main_next_elements, aux_next_elements]
```

Reply:
That's correct, will add a comment explaining that.